packages/app/src/lib/import.ts

```diff
@@ -102,16 +109,90 @@ export async function isCSV(data: string): Promise<Boolean> {
     return papa.parse(data).errors.length === 0;
 }
 
-export async function asCSV(file: File, nameColIndex?: number, tagsColIndex?: number): Promise<VaultItem[]> {
+export async function asCSV(
+    file: File,
+    mappedItemColumns: ImportCSVColumn[],
+    dataOnFirstRow: boolean
+): Promise<{ items: VaultItem[]; itemColumns: ImportCSVColumn[] }> {
     const data = await readFileAsText(file);
     const papa = await loadPapa();
     const parsed = papa.parse(data);
     if (parsed.errors.length) {
         throw new Err(ErrorCode.INVALID_CSV, "Failed to parse .csv file.");
     }
-    return fromTable(parsed.data, nameColIndex, tagsColIndex);
-}
+    const rows = parsed.data as string[][];
+
+    if (rows.length === 0) {
+        throw new Err(ErrorCode.INVALID_CSV, "No rows found in .csv file.");
+    }
 
+    const columnNames = rows[0].map((column) => column.toLowerCase());
+
+    const itemColumns =
+        mappedItemColumns.length > 0
+            ? mappedItemColumns
+            : columnNames.map((columnName, columnIndex) => {
+                  let type = guessFieldType({
+                      name: columnName,
+                      value: rows[1] ? rows[1][columnIndex] : "",
+                  }) as ImportCSVColumn["type"];
+
+                  const lowerCaseColumnName = columnName.toLocaleLowerCase();
+
+                  if (lowerCaseColumnName === "name" || lowerCaseColumnName === $l("name")) {
+                      type = "name";
+                  }
+
+                  if (
+                      lowerCaseColumnName === "tags" ||
+                      lowerCaseColumnName === $l("tags") ||
+                      lowerCaseColumnName === "category" ||
+                      lowerCaseColumnName === $l("category")
```

Comment:
For conditionals like this I generally prefer 

```ts
["tags", ...].includes(lowerCaseColumnName).
```
but that's just a matter of preference I guess.

Reply:
Also, I'm not sure if it's worth comparing it to the translated value, since there is a fairly low chance of the translation just happening to be what the author of the file chose for the tags/category column. If you do, though, you should probably `.toLowerCase()` the translation.